Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -140,6 +143,14 @@
   public long maxMillisToWaitForServerResponse = 0L;
   @NotNull private final InteractiveErrorReporter myErrorReporter = new InteractiveErrorReporter();
 
+  @NotNull private final Object myDebugLogLock = new Object();
+  @NotNull private final LinkedHashMap<String, String> myDebugLog = new LinkedHashMap<String, String>(DEBUG_LOG_CAPACITY) {
```

Comment:
I must be missing something. The complexity is due to reversing the order of the entries. Changing to a queue doesn't help with that. I don't think it is worth the trouble to rewrite.